tempto-core/src/main/java/com/teradata/tempto/internal/query/QueryResultValueComparator.java

```diff
@@ -183,14 +184,24 @@ private int booleanEqual(Object actual, Object expected)
 
     private int longEqual(Object actual, Object expected)
     {
-        if (!(isIntegerValue(actual) && isIntegerValue(expected))) {
+        if (!(actual instanceof Number && expected instanceof Number)) {
```

Comment:
Fair. But you should at least limit this to integers, not all numbers.
So Long, Integer, Short, Byte are the only ones accepted.